test/e2e/helpers/channel_defaulter_test_helper.go

```diff
@@ -0,0 +1,178 @@
+	channelName := "e2e-defaulter-channel"
+	senderName := "e2e-defaulter-sender"
+	subscriptionName := "e2e-defaulter-subscription"
+	loggerPodName := "e2e-defaulter-logger-pod"
+
+	// create channel
+	client.CreateDefaultChannelOrFail(eventingtesting.NewChannel(channelName, client.Namespace))
+
+	// create logger service as the subscriber
+	pod := resources.EventLoggerPod(loggerPodName)
+	client.CreatePodOrFail(pod, common.WithService(loggerPodName))
+
+	// create subscription to subscribe the channel, and forward the received events to the logger service
+	client.CreateSubscriptionOrFail(
+		subscriptionName,
+		channelName,
+		common.ChannelTypeMeta,
+		resources.WithSubscriberForSubscription(loggerPodName),
+	)
+
+	// wait for all test resources to be ready, so that we can start sending events
+	if err := client.WaitForAllTestResourcesReady(); err != nil {
+		t.Fatalf("Failed to get all test resources ready: %v", err)
+	}
+
+	// check if the defaultchannel creates exactly one underlying channel given the spec
+	metaResourceList := resources.NewMetaResourceList(client.Namespace, common.GetChannelTypeMeta(expectedChannelKind))
+	objs, err := base.GetGenericObjectList(client.Dynamic, metaResourceList, &eventingduck.SubscribableType{})
+	if err != nil {
+		t.Fatalf("Failed to list the underlying channels: %v", err)
+	}
+	if len(objs) != 1 {
+		t.Fatalf("The defaultchannel is expected to create 1 underlying channel, but got %d", len(objs))
+	}
+
+	// send fake CloudEvent to the channel
+	body := fmt.Sprintf("TestSingleEvent %s", uuid.NewUUID())
+	event := &resources.CloudEvent{
+		Source:   senderName,
+		Type:     resources.CloudEventDefaultType,
+		Data:     fmt.Sprintf(`{"msg":%q}`, body),
+		Encoding: resources.CloudEventDefaultEncoding,
+	}
+
+	if err := client.SendFakeEventToAddressable(senderName, channelName, common.ChannelTypeMeta, event); err != nil {
+		t.Fatalf("Failed to send fake CloudEvent to the channel %q", channelName)
+	}
+
+	// verify the logger service receives the event
+	if err := client.CheckLog(loggerPodName, common.CheckerContains(body)); err != nil {
+		t.Fatalf("String %q not found in logs of logger pod %q: %v", body, loggerPodName, err)
+	}
+}
+
+// updateDefaultChannelCM will update the default channel configmap
+func updateDefaultChannelCM(client *common.Client, updateConfig func(config *defaultchannel.Config)) error {
+	systemNamespace := resources.SystemNamespace
+	cmInterface := client.Kube.Kube.CoreV1().ConfigMaps(systemNamespace)
+	// get the defaultchannel configmap
+	configMap, err := cmInterface.Get(configMapName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	// get the defaultchannel config value
+	defaultChannelConfig, hasDefault := configMap.Data[channelDefaulterKey]
+	config := &defaultchannel.Config{}
+	if hasDefault {
+		if err := yaml.Unmarshal([]byte(defaultChannelConfig), config); err != nil {
```

Comment:
look at the channelDefaulter.UpdateConfigMap... we are converting the yaml to json, and then doing some update there... I think there was a problem with the casing of the keys otherwise... 
BTW, is that something we can call, instead of duplicating the logic here? If not, no worries...

Reply:
We'll have to update the `configmap` ref first, then call the `UpdateConfigMap` method, and seems we don't have existed code doing the former. 
And I feel the channelDefaulter is more for internal implementation, the only thing interested to end user is the configmap itself.
So I didn't change anything here..